res/app/components/stf/tokens/access-token-service.js

```diff
@@ -0,0 +1,35 @@
+module.exports = function AccessTokenServiceFactory(
+  $rootScope
+, $http
+, socket
+) {
+  var AccessTokenService = {}
+
+  AccessTokenService.getAccessTokens = function() {
+    return $http.get('/app/api/v1/accessTokens')
```

Comment:
I think it's OK if it's easier to leave it like this, since it at least ensures that the data will always be up to date, and it will only get called when the user is working with the API key UI. You can do it as an exercise if you want to, though :)